src/components/Carousel/Carousel.jsx

```diff
@@ -0,0 +1,239 @@
+  justify-content: center;
+`;
+
+const IndicatorListItem = styled.li`
+  list-style: none;
+  padding: 0 4px;
+`;
+
+const IndicatorButton = styled.button`
+  width: 13px;
+  height: 13px;
+  border: none;
+  border-radius: 50%;
+  background-color: ${props => (props.active ? white : gray)};
+
+  :hover {
+    cursor: pointer;
+  }
+`;
+
+const Content = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+class Carousel extends React.Component {
+  state = {
+    left: 0,
+    currentSlideIndex: 0,
+  };
+
+  moveBackward = () => {
+    const { left } = this.state;
+    const { width } = this.props;
+    if (left === 0) {
+      // if it's the first item we still want to update the currentSlideIndex so we can
+      // update the aria-hidden attribute appropriately
+      this.setState(prevState => ({
+        currentSlideIndex:
+          prevState.currentSlideIndex === 0
+            ? prevState.currentSlideIndex
+            : prevState.currentSlideIndex - 1,
+      }));
+      return;
+    }
+    this.setState(prevState => ({
+      left: prevState.left + parseInt(width, 10),
+      currentSlideIndex: prevState.currentSlideIndex - 1,
+    }));
+  };
+
+  moveForward = () => {
+    const { children, width } = this.props;
+    if (this.verifyLastItem(children.length, width)) {
+      // if it's the last item we still want to update the currentSlideIndex so we can
+      // update the aria-hidden attribute appropriately
+      this.setState(prevState => ({
+        currentSlideIndex:
+          prevState.currentSlideIndex === children.length - 1
+            ? prevState.currentSlideIndex
+            : prevState.currentSlideIndex + 1,
+      }));
+      return;
+    }
+
+    this.setState(prevState => ({
+      left: prevState.left - parseInt(width, 10),
+      currentSlideIndex: prevState.currentSlideIndex + 1,
+    }));
+  };
+
+  verifyLastItem = (lengthOfChildren, widthOfEachItem) => {
+    const { left } = this.state;
+    const finalLength = (lengthOfChildren - 1) * parseInt(widthOfEachItem, 10);
+
+    return left === -finalLength;
+  };
+
+  renderItems = () => {
+    const { currentSlideIndex } = this.state;
+    const { children } = this.props;
+    const items = children.map((child, index) => (
+      <MainListItem key={index} aria-hidden={currentSlideIndex !== index}>
+        {child}
+      </MainListItem>
+    ));
+
+    return items;
+  };
+
+  renderAnnouncement = () => {
+    const { currentSlideIndex } = this.state;
+    const { children } = this.props;
+    const content = (
+      <Announcement aria-live="polite" aria-atomic="true" tabIndex={-1}>
+        Slide
+        {currentSlideIndex + 1}
+        out of
+        {children.length}
+      </Announcement>
+    );
+
+    return content;
+  };
+
+  goToSlide = index => {
+    const { width } = this.props;
+    const newLeft = parseInt(width, 10) * index;
+    this.setState({
+      left: -newLeft,
+      currentSlideIndex: index,
+    });
+  };
```

Comment:
Really keen to use only this function to move back and forth like,

`onClick={() => this.goToSlide(currentSlideIndex +/- 1}`

but I wasn't sure how to do those checks that I have in `moveForward` and `MoveBackwards` that let us check if it's the first or last item and stop moving the carousel.

Would love some ideas if you have them, even if it's to keep the functions as they are. I might be over-engineering this

Reply:
you should have access to the current slide using `this.state.currentSlideIndex` so you should be able to determine whether you are moving forwards or backwards vs the `index` which is passed into the function. So knowing that, you can then say `this.state.currentSlideIndex - index > 0` for movingForward, or `this.state.currentSlideIndex - index < 0` for moving backwards, or 0 for already there.
Then you can do those same checks in 1 place